subprojects/core/src/main/java/org/gradle/api/internal/file/copy/DefaultFileCopyDetails.java

```diff
@@ -218,13 +220,19 @@ public ContentFilterable expand(Map<String, ?> properties) {
     @Override
     public void setDuplicatesStrategy(DuplicatesStrategy strategy) {
         this.duplicatesStrategy = strategy;
+        this.defaultDuplicatesStrategy = strategy == DuplicatesStrategy.INHERIT;
```

Comment:
Why do we store this instead of calculating it when needed?

Reply:
When the default duplicates strategy is used, then this class is initialized with the duplicates strategy `INCLUDE`. Only when the duplicates strategy is set on the `FileCopyDetails` later on we need to update whether the default strategy is still used.